### PR TITLE
Copy model before benchmark warmup runs

### LINKED ISSUE #144772: aot_inductor TIMM convit_base inference regression on dashboard
See https://hud.pytorch.org/benchmark/timm_models/inductor_aot_inductor?dashboard=torchinductor&startTime=Tue,%2031%20Dec%202024%2015:26:32%20GMT&stopTime=Tue,%2014%20Jan%202025%2015:26:32%20GMT&granularity=hour&mode=inference&dtype=bfloat16&deviceName=cuda%20(a100)&lBranch=main&lCommit=1dab79470dbecef79ba4c7d4308d8a181091e58e&rBranch=main&rCommit=01034e963c9102c6a4a666c7666afd12aee0bfb3

The model reports a pass but the speedup numbers have gone to 0, which might imply that something went wrong in the reporting process?

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @desertfire @chenyang78 @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4 @huydhn

### PR DESCRIPTION
**Pull request by angelayi**
Fixes https://github.com/pytorch/pytorch/issues/144772

The eager warmup runs causes the model to change state so that later when we export it, the model is different than when we export it directly out of box. For some reason exporting the model with the changed state causes issues but exporting the inital model is ok. This is the reason why the accuracy checks pass but the performance check fails when exporting. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by desertfire — APPROVED** [review]
Similar to the observation in https://github.com/pytorch/pytorch/issues/113029#issuecomment-1795853724

**Comment by StrongerXi** [comment]
I think this causes issue with fsdp because we can't deepcopy it:

```
$ python ./benchmarks/dynamo/torchbench.py --performance --training --fsdp --only torchrec_dlrm
```

Error: P1727060857.

Is the deepcopy only needed under some export setting? If so can we condition it under that? cc @bdhirsh

**Comment by angelayi** [comment]
@StrongerXi the model fails when exporting w/o deepcopying because the model state changes in a previous run. I guess you could gatekeep it for only in export, but it seems a little weird that the model isn't being copied to begin with, because the other benchmarking would be on a modified model.
